src/app/pages/media-details/media-details.component.ts

```diff
@@ -1,28 +1,40 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, switchMap } from 'rxjs';
 import { DetailsMediaService } from 'src/app/features/media/services/details-media.service';
 import { DetailsMediaData, MediaResult } from 'src/app/features/media/types';
 
+const DEFAULT_DATA: DetailsMediaData = {type: 'movie', id: 123 }
 
 @Component({
   selector: 'mov-media-details',
   templateUrl: './media-details.component.html',
   styleUrls: ['./media-details.component.scss'],
 })
-export class MediaDetailsComponent {
+export class MediaDetailsComponent implements OnInit {
   constructor(
     private readonly apiDetailsMedia: DetailsMediaService,
+    private readonly route: ActivatedRoute
   ) {
+  }
 
-    this.apiDetailsMedia.requestDetailsByType<MediaResult>(this.data)
-      .subscribe((detail:MediaResult) => {
-        this.detailData = detail
-        console.log(this.detailData)
-      }
-    )
+  public data:DetailsMediaData = {
+    type: DEFAULT_DATA.type,
+    id: DEFAULT_DATA.id
   }
+  public mediaDetails$?: Observable<MediaResult>
 
-  @Input() public data:DetailsMediaData = {type: 'movie', id: 123 }
 
-  public detailData?:MediaResult
+  public ngOnInit(): void {
+    this.mediaDetails$ = this.route.paramMap.pipe(
+      switchMap(params => {
+        this.data.id = Number(params.get('id')) || DEFAULT_DATA.id
+        return this.getMediaDetails(this.data)
+      })
+    )
```

Comment:
This assignment can be moved from the hook, so the field 'mediaDetails$' will be not optional.

Also it is highly important to unsubscribe from observables when component or directive is destroyed to avoid at least memory leaks.